pennylane/fourier/utils.py

```diff
@@ -67,7 +67,8 @@ def get_spectrum(op, decimals):
         if isinstance(g, np.ndarray):
             matrix = g
         elif hasattr(g, "matrix"):
-            matrix = g.matrix
+            # TODO [Maria]: if generators are class instances, change to matrix()
+            matrix = g.compute_matrix()
```

Comment:
This whole if statement will be simplified with the generator change coming up 🙌

Reply:
wait, shouldn't this take arguments?

Reply:
Yes and no. I think at the moment all generators in the code base are Paulis and don't take args? 

This is a pattern I want to avoid in future: automatically calling class methods. Is there any hotfix we can use for now?

Reply:
> all generators in the code base are Paulis and don't take args?

Oh of course!

Reply:
I added a better comment